src/schema/validator.ts

```diff
@@ -0,0 +1,174 @@
+    }
+    if (conditions.client_filters !== undefined) {
+        validateClientFilters(conditions.client_filters);
+    }
+}
+
+function validateClientFilters(client_filters: any) {
+    if (!Array.isArray(client_filters)) {
+        throw new TypeError("'client_filters' must be an array.");
+    }
+
+    for (const filter of client_filters) {
+        if (typeof filter.name !== "string") {
+            throw new TypeError("Client filter 'name' must be a string.");
+        }
+        if (filter.parameters !== undefined && typeof filter.parameters !== "object") {
+            throw new TypeError("Client filter 'parameters' must be an object.");
+        }
+    }
+}
+
+function validateVariants(variants: any) {
+    if (!Array.isArray(variants)) {
+        throw new TypeError("'variants' must be an array.");
+    }
+
+    for (const variant of variants) {
+        if (typeof variant.name !== "string") {
+            throw new TypeError("Variant 'name' must be a string.");
+        }
+        // skip configuration_value validation as it accepts any type
+        if (variant.status_override !== undefined && typeof variant.status_override !== "string") {
+            throw new TypeError("Variant 'status_override' must be a string.");
+        }
+        if (variant.status_override !== undefined && variant.status_override !== "None" && variant.status_override !== "Enabled" && variant.status_override !== "Disabled") {
+            throw new TypeError("Variant 'status_override' must be 'None', 'Enabled', or 'Disabled'.");
+        }
+    }
+}
+
+function validateVariantAllocation(allocation: any) {
+    if (typeof allocation !== "object") {
+        throw new TypeError("Variant 'allocation' must be an object.");
+    }
+
+    if (allocation.default_when_disabled !== undefined && typeof allocation.default_when_disabled !== "string") {
+        throw new TypeError("Variant allocation 'default_when_disabled' must be a string.");
+    }
+    if (allocation.default_when_enabled !== undefined && typeof allocation.default_when_enabled !== "string") {
+        throw new TypeError("Variant allocation 'default_when_enabled' must be a string.");
+    }
+    if (allocation.user !== undefined) {
+        validateUserVariantAllocation(allocation.user);
+    }
+    if (allocation.group !== undefined) {
+        validateGroupVariantAllocation(allocation.group);
+    }
+    if (allocation.percentile !== undefined) {
+        validatePercentileVariantAllocation(allocation.percentile);
+    }
+    if (allocation.seed !== undefined && typeof allocation.seed !== "string") {
+        throw new TypeError("Variant allocation 'seed' must be a string.");
+    }
+}
+
+function validateUserVariantAllocation(UserAllocations: any) {
+    if (!Array.isArray(UserAllocations)) {
+        throw new TypeError("User allocation 'user' must be an array.");
+    }
+
+    for (const allocation of UserAllocations) {
+        if (typeof allocation.variant !== "string") {

```

Comment:
Can we add a check whether `allocation` is an object? To avoid the following wrong usage:

``` json
"allocaton": {
  "user": [
     "usera", "userb", "userc"
   ]
}
```

Currently, the above usage will get the error message "User allocation 'variant' must be a string."

Reply:
Same suggestion for group and percentile allocation.